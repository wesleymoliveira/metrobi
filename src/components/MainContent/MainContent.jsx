import { Typography } from '../Typography/Typography';
import * as S from './MainContent.styled';

export function MainContent() {
	return (
		<S.Wrapper data-testid='main-content'>
			<S.Content>
				<Typography size='xxlarge'>
					Main Content
				</Typography>
				<Typography as="span">
      	  ** If thing do not look right, make sure your browser <br /> is in
      	  "Experimental Mode".
				</Typography>
			</S.Content>
		</S.Wrapper>
	);
}
