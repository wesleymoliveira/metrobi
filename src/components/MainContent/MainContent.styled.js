import styled, { css } from "styled-components";

export const Wrapper = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.mainContentBackground};
    flex: 5;
    display: flex;
    justify-content: center;
    align-items: start;
  `}
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-items: center;
  text-align: center;
`;
