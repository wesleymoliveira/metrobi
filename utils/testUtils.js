import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
