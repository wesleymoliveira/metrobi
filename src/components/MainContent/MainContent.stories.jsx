import React from 'react';
import { MainContent } from './MainContent';

const MainContentArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },

    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/MainContent',
	component: MainContent,
	argTypes: MainContentArgTypes
};

export const Default = (args) => <MainContent {...args} />;
