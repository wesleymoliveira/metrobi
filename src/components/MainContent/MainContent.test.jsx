import { screen } from '@testing-library/react';

import { MainContent } from './MainContent';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<MainContent />', () => {
		it('should render the component', () => {
		render(<MainContent />);
		
		expect(screen.getByTestId(/main-content/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<MainContent />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<MainContent />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#f5c632')
	});

});