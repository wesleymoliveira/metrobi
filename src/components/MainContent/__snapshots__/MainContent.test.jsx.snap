// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContent /> should match snapshot 1`] = `
.c2 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.6rem;
}

.c0 {
  background-color: #f5c632;
  -webkit-flex: 5;
  -ms-flex: 5;
  flex: 5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  justify-items: center;
  text-align: center;
}

<section
  className="c0"
  data-testid="main-content"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
      color="white"
      data-testid="typography"
      size="xxlarge"
    >
      Main Content
    </h1>
    <span
      className="c3"
      color="white"
      data-testid="typography"
      size="medium"
    >
      ** If thing do not look right, make sure your browser 
      <br />
       is in "Experimental Mode".
    </span>
  </div>
</section>
`;
